### PR TITLE
feat(TCOMP-481): use Definition#getImagePath(type) for all image resources.

### PR DESCRIPTION
**Pull request by RyanSkraba**
**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

There are several ways to get image information and define resource types:

* `ComponentWizardDefinition#getPngImagePath` and `WizardImageType`
* `ComponentDefinition#getPngImagePath` and `ComponentImageType`
* `Definition#getImagePath` (no resource types).

**What is the new behavior?**

These should all be deprecated and use the daikon `Definition#getImagePath(type)` with `DefinitionImageType`.

This relies on https://github.com/Talend/daikon/pull/143

**Does this PR introduce a breaking change?**

- [X] Yes
- [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

All existing definitions need to add the new methods for image resource discovery.

This can be done for all components in a backwards compatible manner in AbstractComponentDefinition, which needs to be prepared for the components repository.

**Other information**:

The `getIconKey()` has been added for components that can supply a standardized "token" to use an image directly from the Talend/ui repository.  This is optional and can return null if no standardized icon is available or planned.

Some documentation updates to clarify the use of image resources for the Definitions.

### CODE REVIEW

File: components/components-elasticsearch/elasticsearch-definition/src/main/java/org/talend/components/elasticsearch/ElasticsearchDatasetDefinition.java [hunk 1]
```diff
@@ -53,9 +54,23 @@ public RuntimeInfo getRuntimeInfo(ElasticsearchDatasetProperties properties) {
         }
     }
 
+    @Deprecated
     @Override
     public String getImagePath() {
         return NAME + "_icon32.png";
     }
 
+    @Override
```
Comment (pbailly) [inline components/components-elasticsearch/elasticsearch-definition/src/main/java/org/talend/components/elasticsearch/ElasticsearchDatasetDefinition.java]: Can't we jsut use the one from AbstractComponentDefinition ?

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/961/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/961/)

File: components/components-elasticsearch/elasticsearch-definition/src/main/java/org/talend/components/elasticsearch/ElasticsearchDatasetDefinition.java (hunk 1 above)
Comment (RyanSkraba) [inline components/components-elasticsearch/elasticsearch-definition/src/main/java/org/talend/components/elasticsearch/ElasticsearchDatasetDefinition.java]: Sorry -- this doesn't inherit from AbstractComponentDefinition!

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/992/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/992/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/994/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/994/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/996/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/996/)

**Pull request merged**